DirectEmployers/MyJobs
# [PD-3445] Update Report Generation Logic to Honor Impersonation

## What's this PR do?
Darrin's updates to the way we create/handle reports did not honor the impersonation logic for analytics. This PR add a specialized user.can call to filter out activities that are to be ignored for staff.

## Screenshots (if appropriate)

## Where should the reviewer start?
* Change your company to a company with the Impersonate app access.
* Ensure you are an admin/are assigned to a role with the "impersonate analytics" activity
* Head to Analytics
## How should this be manually tested?
* First, ensure that the data displayed is correct for the company you are signed in as.
* Second, change the company in the textbox in the upper left corner.
* Finally, test the analytics platform (including exports) and ensure the data matches the impersonated company.
## Any background context you want to provide?
This ticket is a tricky one, because we have to be careful not to grant permission to staff to view protected information (such as PRM reports). This approach is designed to avoid that type of unintended access.

**Note, DNM Label added until I can add proper tests**
